### PR TITLE
rgw: persist rgw_lc_max_objs

### PR DESCRIPTION
**Pull request by cooboos**
This argument(rgw_lc_max_objs) is not persisted, if it's changed to
a smaller value, some bucket's lifecycle config will be invalid.

Signed-off-by: wangsongbo wangsongbo@cloudin.cn

### CODE REVIEW

File: src/rgw/rgw_lc.cc [hunk 1]
```diff
@@ -171,16 +171,53 @@ void *RGWLC::LCWorker::entry() {
...
 
-  for (int i = 0; i < max_objs; i++) {
+    ret = cls_rgw_lc_get_meta(store->lc_pool_ctx, lc_meta_obj_name, lc_objs, meta);
+    if (ret) {
+      ldout(cct, 0) << "RGWLC::initialize() failed to get lc meta " << lc_meta_obj_name << dendl;
+      l.unlock(&store->lc_pool_ctx, lc_meta_obj_name);
+      return ret;
+    } else {
+      if (meta.second >= max_objs) {
+	objs = meta.second;
```
Comment (ZVampirEM77) [inline src/rgw/rgw_lc.cc]: indent

**Comment by ZVampirEM77** [comment]
@cooboos if I understand correctly, for persisting rgw_lc_max_objs, you record it as a rados obj in lc pool, right? And the case `some bucket's lifecycle config will be invalid` will happen if I configure the `rgw_lc_max_objs` to a smaller value, then restart RGW, right?

But I do not think this is a good implement. In this implement, a smaller `rgw_lc_max_objs` will never be set, right? It is to say, users can not configure a smaller `rgw_lc_max_objs`, even though they really want to do it.

**Comment by cooboos** [comment]
@ZVampirEM77  Yes, it's that you understand. I  agree with you. I will modify this and push it later.

**Comment by ZVampirEM77** [comment]
@cooboos could you please add a tracker for us to trace this issue and make a backport if necessary easily~ thank you~

**Comment by cooboos** [comment]
@ZVampirEM77   [tracker22246](http://tracker.ceph.com/issues/22246)

**Comment by cooboos** [comment]
@ZVampirEM77 ， I've made some changes:
1. Record the previous value(`rgw_lc_max_objs`) as a rados obj(`lc.meta`) in lc pool 
2.Add a `reset_all_entry` function  that can reset all the bucket entry, when the new value is not equal  to the recorded. So a bigger or smaller value can be set .
3. Add a `restore_miss_entry`  function that  can recovery the bucket entries, when  RGW start failed for some abnormal case.
check pls.

**Comment by mattbenjamin** [comment]
@dang thoughts?

**Pull request closed without merging**